includes/abstracts/abstract-wc-stripe-payment-gateway.php

```diff
@@ -157,7 +157,9 @@ public function is_retryable_error( $error ) {
 			(
 				'payment_intent_mandate_invalid' === $error->code || // Don't retry payments when a 3DS mandate is invalid.
 				'charge_exceeds_transaction_limit' === $error->code || // Don't retry payments when the charge exceeds the transaction limit.
-				'amount_too_small' === $error->code
+				'amount_too_small' === $error->code ||
+				'card_declined' === $error->code ||
+				'payment_method_provider_decline' === $error->code
 			)
```

Comment:
I agree with Diego's comment here about using `in_array()`, especially now that we're up to 5 candidate values!
```suggestion
			in_array(
				$error->code,
				[
					'payment_intent_mandate_invalid', // Don't retry payments when a 3DS mandate is invalid.
					'charge_exceeds_transaction_limit', // Don't retry payments when the charge exceeds the transaction limit.
					'amount_too_small',
					'card_declined',
					'payment_method_provider_decline',
				],
				true
			)

Reply:
Note: this is also a nit, and not blocking!

Reply:
Refactored this function in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4758/commits/2febcbf36ba4c59214744815f80619559364f5bf